### PR TITLE
[k8s-configuration] Update extension CLI to v2.1.0

### PR DESCRIPTION
**Pull request by bavneetsingh16**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/k8s-configuration/azext_k8s_configuration/_params.py [hunk 1]
```diff
@@ -320,6 +320,12 @@ def load_arguments(self, _):
             arg_type=get_three_state_flag(),
             help="Re-create resources that cannot be updated on the cluster (i.e. jobs)",
         )
+        c.argument(
+            "disable_health_check",
+            options_list=["--disable-health-check"],
```
Comment (zhoxing-ms) [inline src/k8s-configuration/azext_k8s_configuration/_params.py:325]:
```suggestion
```

**Comment by zhoxing-ms** [comment]
Could you please add some test cases for this new feature?

**Comment by bavneetsingh16** [comment]
> Could you please add some test cases for this new feature?

We have set up live tests in our forked repository as a continuous integration (CI) check and have included tests for this new parameter. All the tests in the file linked below pertain to the changes in this pull request:
[https://github.com/AzureArcForKubernetes/k8s-configuration/blob/main/testing/test/configurations/FluxKustomization.Wait.Tests.ps1](https://github.com/AzureArcForKubernetes/k8s-configuration/blob/main/testing/test/configurations/FluxKustomization.Wait.Tests.ps1)

As the code owners of this extension CLI, we mandate that all changes be first made in our forked repository to ensure they undergo our comprehensive test suite

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**